src/inmanta/server/services/orchestrationservice.py

```diff
@@ -943,7 +953,17 @@ async def _put_version(
                         raise BadRequest(msg)
                     all_ids |= {Id.parse_id(rid, version) for rid in rids_unchanged_resource_sets.keys()}
 
-                await data.Resource.insert_many(list(rid_to_resource.values()), connection=connection)
+                updated_resources = list(rid_to_resource.values())
+                await data.Resource.insert_many(updated_resources, connection=connection)
+                # bump all resource sets if we are doing a full compile otherwise, bump only the updated resource sets
+                if not is_partial_update:
+                    updated_resource_sets = {r.resource_set if r.resource_set is not None else "" for r in updated_resources}
```

Comment:
Shouldn't this be:
```suggestion
                if not is_partial_update:
                    updated_resource_sets = {r if r is not None else "" for r in resource_sets.values()}
```

Reply:
I believe `resource_sets.values()` will not contain the default resource set